lib/cli/interactive-setup/console-dev-mode.js

```diff
@@ -0,0 +1,337 @@
+          process.stdout.write(chalk.grey(`${time} • ${resourceName} • Log\n`));
+          try {
+            const parsedBody = JSON.parse(activity.body);
+            if (typeof parsedBody === 'string') {
+              throw new Error('Not a JSON object');
+            }
+            process.stdout.write(chalk.bold(`${JSON.stringify(parsedBody, null, 2)}\n`));
+          } catch (error) {
+            process.stdout.write(chalk.bold(`${activity.body}\n`));
+          }
+          break;
+        case 'span': {
+          const span = formatSpan(activity);
+          process.stdout.write(chalk.grey(`${time} • ${resourceName} • Span • ${span.niceName}\n`));
+          break;
+        }
+        case 'aws-lambda-request':
+          process.stdout.write(chalk.grey(`${time} • ${resourceName} • Invocation Started\n`));
+          break;
+        case 'aws-lambda-response':
+          process.stdout.write(chalk.grey(`${time} • ${resourceName} • Invocation Ended\n`));
+          break;
+        case 'event': {
+          const { message, payload } = formatEvent(activity);
+          process.stdout.write(chalk.grey(`${time} • ${resourceName} • ${message}\n`));
+          process.stdout.write(chalk.bold(`${JSON.stringify(payload, null, 2)}\n`));
+          break;
+        }
+        default:
+      }
+    }
+  } catch (error) {
+    process.stdout.write(error, '\n');
+  }
+};
+
+const connectToWebSocket = async (functionName, org) => {
+  const { token } = await apiRequest(`/api/identity/orgs/${org.orgId}/token`);
+  const ws = new WebSocket(`${socketUrl}?Auth=${token}`, {
+    perMessageDeflate: false,
+  });
+
+  ws.on('open', () => {
+    writeText(style.aside('Waiting for dev mode activity...\n'));
+    ws.send(JSON.stringify({ filters: { functionName } }));
+  });
+  ws.on('message', (data) => {
+    streamBuff.put(`${data.toString('utf-8')};;;`);
+  });
+  return ws;
+};
+
+module.exports = {
+  async isApplicable(context) {
+    const { isConsole, launchDev, deployNeeded } = context;
+
+    if (!isConsole) {
+      context.inapplicabilityReasonCode = 'NON_CONSOLE_CONTEXT';
+      return false;
+    }
+
+    if (!launchDev) {
+      context.inapplicabilityReasonCode = 'NON_LAUNCH_DEV_CONTEXT';
+      return false;
+    }
+
+    if (!context.serverless) {
+      context.inapplicabilityReasonCode = 'NO_SERVERLESS_CONTEXT';
+      return false;
+    }
+
+    if (deployNeeded) {
+      context.inapplicabilityReasonCode = 'NEEDS_DEPLOY';
+      return false;
+    }
+
+    if (!(await resolveAuthMode())) {
+      context.inapplicabilityReasonCode = 'NOT_LOGGED_IN';
+      return false;
+    }
+
+    if (!context.org) {
+      context.inapplicabilityReasonCode = 'UNRESOLVED_ORG';
+      return false;
+    }
+
+    return true;
+  },
+
+  async run(context) {
+    streamBuff.on('data', handleSocketMessage);
+    const ws = await connectToWebSocket(context.targetFunctions, context.org);
+    // Only open for 5 minutes
+    setTimeout(() => {
```

Comment:
Ahh also sorry I just realized that I need to add the still working logic. This will not stay fixed as a 5 minute limit.

Reply:
@medikoo Okay I added some logic here to do a few things.

- Reconnect to our websocket every hour. (There is a 2 hour limit on connection in API Gateway so I figured I would reconnect well before this time limit ends)
- After 1.5 hours of inactivity pause the websocket and prompt the user to confirm they are still working. (This is so if someone leaves their computer open we won't just keep their websocket connection alive forever)

[The logic I have right now](https://github.com/serverless/serverless/pull/11782/commits/8549bf16ab60e3aa470449004e792a6b5ed982f0) is a bit confusing so if you have suggestions for improvements please let me know 🙏